lib/grape/formatters/active_model_serializers.rb

```diff
@@ -2,14 +2,33 @@
 #
 # Serializer options can be passed as a hash from your Grape endpoint using env[:active_model_serializer_options],
 # or better yet user the render helper in Grape::Helpers::ActiveModelSerializers
+
+require 'active_model_serializers/serialization_context'
+
 module Grape
   module Formatters
     module ActiveModelSerializers
       def self.call(resource, env)
-        serializer_options = {}
-        serializer_options.merge!(env[:active_model_serializer_options]) if env[:active_model_serializer_options]
+        serializer_options = build_serializer_options(env)
         ::ActiveModelSerializers::SerializableResource.new(resource, serializer_options).to_json
       end
+
+      protected
```

Comment:
public only has to do with someone else calling it.  a public class method can call a private class method, so I think that meets your intent